src/search.coffee

```diff
@@ -67,20 +79,19 @@ class Search
 
       # Filtered searches' data lives within the webPages property.
       body = body.webPages if body._type is 'SearchResponse'
-      return callback 'Invalid HTTP response body.' if not body?
+      return callback new Error 'Invalid HTTP response body.' unless body
 
-      # Parse an ID out of result URLs.
+      # Parse an ID out of result URLs for compatibility and duplicate checks.
       invalidId = false
       body.value.forEach (result) ->
         matches = (result.url or result.hostPageUrl).match /&h=([^&]+)/
-        if not matches?
+        if matches
+          result.id = matches[1]
+        else
           invalidId = true
-          return
-
-        result.id = matches[1]
-      return callback 'Unable to parse an ID out of result URL.' if invalidId
+      if invalidId
+        return callback new Error 'Unable to parse an ID out of result URL.'
```

Comment:
I'd recommend tacking on the failing URL. Whoever is using this library can capture the error and feed into whatever they use later for inspection.

For example, our would end up in Sentry and we could see if we screwed up on URL parsing or if the format changed:

```coffee
_.extend new Error('Unable to parse an ID out of result URL'), url: result.url or result.hostPageUrl
```

This is basically just shorthand for:

```coffee
err = new Error 'Unable to parse an ID out of result URL'
err.url = result.url or result.hostPageUrl
```